pkg/activator/net/helpers.go

```diff
@@ -26,6 +26,24 @@ import (
 	"knative.dev/serving/pkg/apis/networking"
 )
 
+// healthyEndpointsWithPort takes an endpoints object and a port name and return the set
+// of addresses that implement this port.
+func healthyAddressesWithPort(endpoints *corev1.Endpoints, portName string) sets.String {
+	ready := sets.NewString()
+
+	for _, es := range endpoints.Subsets {
+		for _, port := range es.Ports {
+			if port.Name == portName {
```

Comment:
Should be break after we found a match, i.e. can the same port happen more than once in the same subset?